rotkehlchen/premium/premium.py

```diff
@@ -228,11 +234,15 @@ def upload_data(
             'API-SIGN': base64.b64encode(signature.digest()),
         })
 
+        tmp_file = io.BytesIO()
```

Comment:
Why do you need to write the data blob to a temporary file? Can't you stream directly from memory?

Reply:
true, I didn't know but as per the docs seems that you can just add the data there. Thanks

Reply:
Neither did I tbh. I just assumed it has to be possible. Otherwise I would consider this a bug in the library